libraries/AP_Compass/AP_Compass_Calibration.cpp

```diff
@@ -127,7 +136,7 @@ void Compass::_cancel_calibration(uint8_t i)
     AP_Notify::events.initiated_compass_cal = 0;
     Priority prio = Priority(i);
 
-    if (_calibrator[prio].running() || _calibrator[prio].get_status() == CompassCalibrator::Status::WAITING_TO_START) {
+    if (_calibrator[prio].running() || _calibrator[prio].get_state().status == CompassCalibrator::Status::WAITING_TO_START) {
```

Comment:
Pre exsisting, but checking for `WAITING_TO_START`is unneeded because `_status` won't ever be in that state. It will immediately enter `RUNNING_STEP_ONE`. It's technically out of scope, but that whole state should be removed, because it's complicating the calibrator.

Reply:
The calibrator does get into `WAITING_TO_START` step when holding for start delay that might be requested at start method call.